packages/components/src/Modal/src/BaseModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { type ResponsiveProp, type StyledComponentProps, useColorSchemeContext, useResponsiveValue, useStyledSystem } from "@hopper-ui/styled-system";
+import clsx from "clsx";
+import { type CSSProperties, type ForwardedRef, forwardRef } from "react";
+import { ModalOverlay, type ModalOverlayProps, type ModalRenderProps, Modal as RACModal, useContextProps } from "react-aria-components";
+
+import { HopperProvider } from "../../HopperProvider/index.ts";
+import { cssModule } from "../../utils/index.ts";
+
+import { BaseModalContext } from "./BaseModalContext.ts";
+
+import styles from "./BaseModal.module.css";
+
+export const GlobalBaseModalCssSelector = "hop-BaseModal";
+
+export interface BaseModalProps extends StyledComponentProps<ModalOverlayProps> {
+    /**
+     * The size of the modal.
+     * @default "md"
+     */
+    size?: ResponsiveProp<"sm" | "md" | "lg" | "xl" | "fullscreen" | "fullscreenTakeover">;
+    /**
+     * Whether the modal has an image.
+     */
+    hasImage?: boolean;
+}
+
+const BaseModal = (props: BaseModalProps, ref: ForwardedRef<HTMLDivElement>) => {
+    [props, ref] = useContextProps(props, ref, BaseModalContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        style,
+        slot,
+        size: sizeProp,
+        children,
+        hasImage,
+        ...otherProps
+    } = ownProps;
+
+    const { colorScheme } = useColorSchemeContext();
+    const size = useResponsiveValue(sizeProp) ?? "md";
+
+    const classNames = (renderProps: ModalRenderProps) => clsx(
+        GlobalBaseModalCssSelector,
+        cssModule(
+            styles,
+            GlobalBaseModalCssSelector,
+            size.toLowerCase(),
+            renderProps.isEntering && "entering",
+            renderProps.isExiting && "exiting",
+            hasImage && "image"
+        ),
+        stylingProps.className,
+        className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...style,
+        ...stylingProps.style
+    };
+
+    return (
+        <ModalOverlay
+            className={classNames}
+            style={mergedStyles}
+            slot={slot}
+            {...otherProps}
```

Comment:
i would expect otherProps to be set in the RAC Modal, and have an extra prop named `overlayProps` that would be forwarded to the overlay